README.md

```diff
@@ -0,0 +1,197 @@
+
+9. Drop the FPGAHealthAndStatus/HealthAndStatusFIFOCopy.vi. Set the internal
+   health and status FIFO. Set the source as the FIFO generated in step 2.
+
+10. Drop the FPGAHealthAndStatus/HealthAndStatusFIFOCopy.vi. Set the internal
+    health and status FIFO. Set the source as the FIFO generated in step 5.
+
+# Theory of operation
+
+FIFOs are used for what they are best suited - to provide a channel between
+variable speeds write and read ends. Usually one end of the FIFO is inside
+[SCTL](https://knowledge.ni.com/KnowledgeArticleDetails?id=kA00Z000000P8sWSAS&l=en-US).
+SCTL are used to grant execution of the commands inside FPGA in a single tick.
+
+Link speed is hard-coded to 44 40Mhz ticks per bit, which roughly equal to
+921600 bauds *(actual number is 40M / 44 = 909090.909; but as delays are
+inserted between frames, and frames aren't long, 44 works, sampling a slightly
+different part of receiving signal for each received bit)*.
+
+One start and one stop bits are written (must be included in data, see below).
+
+Transmitting FIFO is U16 typed. The most significant nibble is used for
+instruction. Nibbles 2-4 are used for data.
+
+Data from transmitting FIFO are parsed into instructions and data, and placed
+into internal FIFO. Instructions are then processed from the internal FIFO -
+see below for instructions types. Proper signal (frame start, start bit, data,
+stop bit, frame delay) are generated.
+
+Received line is checked for frame start. Bytes are then stored into receiving
+FIFO. Frame end is detected on line silence. IRQ is raised when frame is
+received. As the expected use is in a master-slave communication, IRQ doesn't
+have to be handled outside ModbusPort subVI. Data are writen into receiving
+FIFO.
+
+CRC16 isn't checked on received FIFO and the CRC16 (if part of the
+communication) is stored into receiving FIFO.
+
+# Instructions
+
+Opcodes 1-8 are used for transmission (Tx FIFO). Instructions 9 and 10 are used
+in received response (Rx FIFO).
+
+ OpCode | Description
+ ------ | -----------
+ 0x1    | sends data. Data are in bits 9-0, and includes start and stop bits. The least significant bit (bit 0) is a start bit, must be 0. 8 data bits follow. Bit 9 is stop  bit, must be 1. To write 8 bit data, shift left (with 0 fill) by 1, or result with 0x0200
+ 0x2    | End of frame, delay in ticks (data). Should be 0xda, resulting in 0x20da written
+ 0x3    | delay for data ticks
+ 0x4    | delay for data microseconds
+ 0x5    | delay for data milliseconds
+ 0x6    | wait for a received frame. Data = timeout in us (microseconds, 1/1M of a second, 40 ticks). Recommended value is 0x3e8 = 1ms.
+ 0x7    | triggers IRQ
+ 0x8    | wait for trigger (ModbusTrigger == true)
+ 0x9    | received data (on received FIFO)
+ 0xA    | received end of frame (on received FIFO)
+
+For synchronization with the other modbus ports, a frame should be prefixed
+with 0x8 command. Data wouldn't be transmitted before ModbusTrigger == true.
```

Comment:
It is not clear to me if the data is held until ModbusTrigger is true, or if data is simply dropped. Assuming the former, please consider "wouldn't be transmitted before..." to "will not be transmitted until...".